File: day1/part1/solution.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func solve(rotations []string) int {
	currentPosition := 50
	zeroCount := 0

	for _, line := range rotations {
		if len(line) < 2 {
			continue
		}

		direction := line[0]
		distance, err := strconv.Atoi(line[1:])
		if err != nil {
			// Skip lines that don't have a valid number
			continue
		}

		if direction == 'R' {
			currentPosition += distance
		} else if direction == 'L' {
			currentPosition -= distance
		}

		// Go's % operator can return negative results, so we ensure it wraps correctly
		currentPosition = (currentPosition%100 + 100) % 100

		if currentPosition == 0 {
			zeroCount++
		}
	}
	return zeroCount
}

func main() {
	inputFilePath := "day1/part1/input1.txt"
	file, err := os.Open(inputFilePath)
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error opening file '%s': %v\n", inputFilePath, err)
		os.Exit(1)
	}
	defer file.Close()

	var rotations []string
	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		rotations = append(rotations, scanner.Text())
	}

	if err := scanner.Err(); err != nil {
		fmt.Fprintf(os.Stderr, "Error reading file: %v\n", err)
		os.Exit(1)
	}

	password := solve(rotations)
	fmt.Printf("The password is: %d\n", password)
}

File: day1/part1/solution.rs
use std::fs::File;
use std::io::{self, BufRead};
use std::path::Path;

fn solve(rotations: Vec<String>) -> i32 {
    let mut current_position = 50;
    let mut zero_count = 0;

    for line in rotations {
        if line.is_empty() {
            continue;
        }

        let direction = line.chars().next().unwrap();
        let distance: i32 = line[1..].parse().unwrap();

        if direction == 'R' {
            current_position += distance;
        } else if direction == 'L' {
            current_position -= distance;
        }

        current_position = (current_position % 100 + 100) % 100;

        if current_position == 0 {
            zero_count += 1;
        }
    }

    zero_count
}

fn main() -> io::Result<()> {
    let path = Path::new("day1/part1/input1.txt");
    let file = File::open(&path)?;
    let lines = io::BufReader::new(file).lines();

    let rotations: Vec<String> = lines.map(|line| line.unwrap()).collect();
    
    let password = solve(rotations);
    println!("The password is: {}", password);

    Ok(())
}